torch/csrc/PyInterpreter.cpp

```diff
@@ -587,28 +587,27 @@ c10::IntArrayRef ConcretePyInterpreterVTable::strides(
   return c10::IntArrayRef(start, len);
 }
 
-static std::vector<int64_t> values_from_buffer(
-    const c10::TensorImpl* self,
-    py::handle values) {
-  c10::TensorImpl* ptr = const_cast<c10::TensorImpl*>(self);
+template <typename T>
+static void attach_array_lifetime_to_tensor(
+    const c10::TensorImpl* tensor,
+    T* array_data,
+    const char* attr_name) {
+  c10::TensorImpl* t_ptr = const_cast<c10::TensorImpl*>(tensor);
   c10::optional<PyObject*> mb_obj =
-      ptr->pyobj_slot()->check_pyobj(getPyInterpreter());
+    t_ptr->pyobj_slot()->check_pyobj(getPyInterpreter());
   TORCH_CHECK(
       mb_obj.has_value(), "Tensor subclass's PyInterpreter has no value");
-
-  py::object os = py::module_::import("torch").attr("overrides");
-  py::function get_buffer =
-      py::reinterpret_borrow<py::function>(os.attr("get_buffer"));
-  auto buffer = get_buffer(py::handle(*mb_obj), values, "size");
-  auto result = THPUtils_unpackLongs(buffer.ptr());
-  return result;
+  auto capsule = py::capsule(array_data, [](void* p) {
+      delete[] reinterpret_cast<T*>(p);
```

Comment:
This is suspicious to me. It is better code structure if the deleter is created at the same spot where the data is allocated, so it is easier to audit that you have lined them up correctly.